### DISCUSSION TITLE
Include documentation about Reverse Proxy for all other applications on Reverse Proxy documentation

### DISCUSSION THREAD

**Discussion by luzfcb** [nextcloud/all-in-one#2779] [Questions] [answered]
### Is your feature request related to a problem? Please describe. 

Currently, I'm running Nextcloud on a home server that doesn't have a public IP, and there's no simple way to do this.

To solve this problem, I bought a domain, rented a VPS server, and created a VPN network that connects my VPS server to my home server. On the VPS server, I installed Caddy Server and configured it with help documentation on the reverse proxy, and everything works fine except Nextcloud Talk.

Nextcloud AIO provides several other applications besides Nextcloud served by Apache on port `11000`, but there is still no documentation on how to use them with a reverse proxy.

The docker `nextcloud-aio-talk` container exposes only the `3478` port, but inside the container, these are the ports:

```bash
/ $ netstat -tulpn | grep LISTEN | sort -u
tcp        0      0 0.0.0.0:8188            0.0.0.0:*               LISTEN      14/janus
tcp        0      0 127.0.0.11:42307        0.0.0.0:*               LISTEN      -
tcp        0      0 127.0.0.1:3478          0.0.0.0:*               LISTEN      17/turnserver
tcp        0      0 127.0.0.1:3479          0.0.0.0:*               LISTEN      17/turnserver
tcp        0      0 127.0.0.1:4222          0.0.0.0:*               LISTEN      15/nats-server
tcp        0      0 172.19.0.10:3478        0.0.0.0:*               LISTEN      17/turnserver
tcp        0      0 172.19.0.10:3479        0.0.0.0:*               LISTEN      17/turnserver
tcp        0      0 :::8081                 :::*                    LISTEN      16/nextcloud-spreed
tcp        0      0 :::8088                 :::*                    LISTEN      14/janus
tcp        0      0 :::8188                 :::*                    LISTEN      14/janus
```

Assuming my Nextcloud Apache 11000 is being exposed to the internet at `nc.mydomain.com`, my plan is to create subdomains for each of the other applications that need to be available online, like:

`s.nc.mydomain.com:443` -> expose STUN
`t.nc.mydomain.com:443` -> expose TURN

In short, that's how Netmaker VPN does the same: https://github.com/gravitl/netmaker/blob/master/docker/Caddyfile

Additionally, Nextcloud AIO Web Setup should provide an option to modify STUN, TURN, and High-performance backend urls, or if not possible, provide via environment variable in Nextcloud AIO master container.

![image](https://github.com/nextcloud/all-in-one/assets/807599/728bc833-ca98-4e33-8bab-50e4969a2864)

### Describe the solution you'd like

Provide complete documentation for ALL applications that Nextcloud AIO installs and that necessarily need to be exposed to the internet via reverse proxy.

**Comment by szaimen** [comment, marked as answer]
Hi, you can simply install your own turn/stun server on any server or use any publicly available one and configure that in the talk admin settings. It will not get overwritten by AIO.

**Reply by luzfcb** [reply]
Hello @szaimen, thanks for the answer, however, it's still not enough, that's because my ticket is all about documentation, and your answer does not solve this unequivocally for all Nextcloud AIO users, as it only explains what to do and not how to do it and implies that the person installing Nextcloud AIO knows in advance what STUN/TURN server and _High-performance backend_ is and how to configure it correctly with Nextcloud, which is not true at first.

Although it explains what to do, it doesn't explain how to do it.

In order to make life easier for anyone using Nextcloud AIO, I seriously think the https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md  documentation really should be expanded to include the reverse proxy configuration for ALL other services required for Nextcloud works correctly in its entirety.

For my specific case, updating this caddy documentation would be enough:

https://github.com/nextcloud/all-in-one/blob/3bdbe7d59e08681656b5cc9a5713d1a6337db817/reverse-proxy.md?plain=1#L96-L102


but I think it would be necessary to expand the settings of Apache, Caddy with ACME DNS-challenge, Citrix ADC VPX / Citrix Netscaler, Cloudflare Tunnel, HaProxy, Nginx, Nginx-Proxy-Manager, Node.js with Express, Synology Reverse Proxy, Traefik 2 , so this is a lot of work and would need help from the experience of many people who use these proxies to report their experiences.


Is my explanation of why I opened this ticket in the first place enough to keep it open until the documentation is updated to include more information on configuring the other reverse proxy services?

I still can't get Nextcloud Talk to work, and I keep getting Nextcloud Talk UI connection errors.

So if this ticket is still open, someone at some point might try to resolve it, be myself if I manage to get a working configuration (I'm still struggling to figure out how to debug this.), could be someone from the Nextcloud team, if you have time and willingness and knowledge to update or it could be someone from the community who wants to contribute with their experience.

**Comment by szaimen** [comment]
> Hello @szaimen, thanks for the answer, however, it's still not enough, that's because my ticket is all about documentation, and your answer does not solve this unequivocally for all Nextcloud AIO users, as it only explains what to do and not how to do it and implies that the person installing Nextcloud AIO knows in advance what STUN/TURN server and _High-performance backend_ is and how to configure it correctly with Nextcloud, which is not true at first.

Yes, this knowledge is required since you have this very complicated setup. For normal setups no further documentation is needed.

> Although it explains what to do, it doesn't explain how to do it.

I've gave you the idea what you need to do in your case. I will not get further into detail as you can easily google how to do the things that I mentioned.

> In order to make life easier for anyone using Nextcloud AIO, I seriously think the https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md documentation really should be expanded to include the reverse proxy configuration for ALL other services required for Nextcloud works correctly in its entirety.

As I said above on normal setups not further configuration is needed.

> For my specific case, updating this caddy documentation would be enough:
> 
> https://github.com/nextcloud/all-in-one/blob/3bdbe7d59e08681656b5cc9a5713d1a6337db817/reverse-proxy.md?plain=1#L96-L102
> 
> but I think it would be necessary to expand the settings of Apache, Caddy with ACME DNS-challenge, Citrix ADC VPX / Citrix Netscaler, Cloudflare Tunnel, HaProxy, Nginx, Nginx-Proxy-Manager, Node.js with Express, Synology Reverse Proxy, Traefik 2 , so this is a lot of work and would need help from the experience of many people who use these proxies to report their experiences.

As I said above on normal setups not further configuration is needed.

> Is my explanation of why I opened this ticket in the first place enough to keep it open until the documentation is updated to include more information on configuring the other reverse proxy services?
> 
> I still can't get Nextcloud Talk to work, and I keep getting Nextcloud Talk UI connection errors.

As I said above, you can simply install your own turn/stun server on any server or use any publicly available one and configure that in the talk admin settings. It will not get overwritten by AIO.

> So if this ticket is still open, someone at some point might try to resolve it, be myself if I manage to get a working configuration (I'm still struggling to figure out how to debug this.), could be someone from the Nextcloud team, if you have time and willingness and knowledge to update or it could be someone from the community who wants to contribute with their experience.

I would recommend to ask on https://help.nextcloud.com if you should need further help on this.

**Reply by luzfcb** [reply]
The current documentation, in my opinion, does not explicitly explain what should be accessible to the internet and what should not, nor how to do it.

I used Google a lot and did a lot of tests before opening this ticket. I still couldn't get anything working for my configuration (which is quite common for[ Unraid OS](https://unraid.net/) users who the internet provider does not allow access and make configurations on the gateway device (ONU GPON/EPON to ethernet converter), do port forwarding, and the IP is provided dynamically and may change without notice).

So, we agree to disagree that the [current documentation](https://github.com/nextcloud/all-in-one/blob/14b598adc4435d4a80b9f4091104e3e2a297a3d5/reverse-proxy.md) is enough to minimally help and guide the Nextcloud AIO user on how to solve their problems on their own, and that's okay.

I'll look for help at https://help.nextcloud.com/, as you suggested.

Anyway, thanks for the hard work of the Nextcloud team in making this amazing piece of software.
